### PR TITLE
feat: implement account linking plugin

### LINKED ISSUE #7556: feat(jans-casa): add account linking plugin
Add the equivalent of Gluu casa acct linking but using Agama

### PR DESCRIPTION
**Pull request by jgomer2001**
### Prepare

- [ ] Read [PR guidelines](https://github.com/JanssenProject/jans/blob/main/docs/CONTRIBUTING.md#prs)
- [ ] Read [license information](https://github.com/JanssenProject/jans/blob/main/LICENSE)

-------------------

#### Target issue
  
  
  
  
closes #7556

#### Implementation Details
  

-------------------
### Test and Document the changes
- [ ] Static code analysis has been run locally and issues have been fixed
- [ ] Relevant unit and integration tests have been added/updated
- [ ] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

### CODE REVIEW

File: docs/casa/plugins/accts-linking/accts-linking-agama.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        "clientSecret": "m|a1l2d3i4t5a6S7h8a9k0i'r¿a",
+        "scopes": ["email", "profile"]
+    }
+}
+   
+```         
+
+In this case, we are populating the configuration of an OAuth-based provider called "Goooogle". 
+
+The tables shown in [this](https://github.com/JanssenProject/jans/blob/main/docs/agama-catalog/jans/inboundID/README.md#supply-configurations) page list all possible properties to configure a provider. Particularly, two properties deserve the most detail:
```
Comment (moabu) [inline docs/casa/plugins/accts-linking/accts-linking-agama.md:39]: This link should be a https://docs.jans.io . @ossdhaval this has to be added to the mkdocs and then referenced here.

File: docs/casa/plugins/accts-linking/accts-linking-agama.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    
+
+2. `mappingClassField`. This is key for performing the attribute mapping process and the user provisioning. The remainder of this document is dedicated to these two aspects
+
+!!! Note
+    Recall `enabled` is a handy property that can be used to temporarily "deactive" a given identity provider.
+
+## Configuring attribute mappings
+
+An introduction to attribute mapping can be found [here](https://github.com/JanssenProject/jans/blob/main/docs/agama-catalog/jans/inboundID/README.md#attribute-mappings). Unless an elaborated processing of attributes is required, a basic knowledge of Java language suffices to write a useful mapping.
```
Comment (moabu) [inline docs/casa/plugins/accts-linking/accts-linking-agama.md:55]: Again the same thing as above 

CC @ossdhaval

File: docs/casa/plugins/accts-linking/index.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
+# Accounts Linking Plugin
+
+## Overview
+
+This plugin allows users to "link" their local Jans account with existing accounts at third-party identity providers like OIDC OPs and social sites, e.g. Apple, Facebook, Google, etc.
+
+Besides the usual onboarding of a plugin jar file in Casa, administrators must deploy a number of additional components. This will be regarded later. By now let's summary key points of the accounts linking experience:
```
Comment (moabu) [inline docs/casa/plugins/accts-linking/index.md]: However, lets summarize the key points of the ....

**Review by moabu — CHANGES_REQUESTED** [review]
@iromli we should be prepared to auto add the jars once merged to our docker prep steps.

**Review by maduvena — APPROVED** [review]

File: docs/casa/plugins/accts-linking/index.md (hunk 3 above)
Comment (jgomer2001) [reply docs/casa/plugins/accts-linking/index.md]: updated

Comment (nynymike) [reply docs/casa/plugins/accts-linking/index.md]: Can we just make issues to track the changes to docs so we can start testing?

**Review by moabu — APPROVED** [review]

**Pull request merged**